flink-examples/flink-examples-streaming/src/main/scala/org/apache/flink/streaming/scala/examples/ml/IncrementalLearningSkeleton.scala

```diff
@@ -0,0 +1,169 @@
+    val env = StreamExecutionEnvironment.getExecutionEnvironment
+    env.setStreamTimeCharacteristic(TimeCharacteristic.EventTime)
+
+    // build new model on every second of new data
+    val trainingData = env.addSource(new FiniteTrainingDataSource)
+    val newData = env.addSource(new FiniteNewDataSource)
+
+    val model = trainingData
+      .assignTimestampsAndWatermarks(new LinearTimestamp)
+      .timeWindowAll(Time.of(5000, TimeUnit.MILLISECONDS))
+      .apply(new PartialModelBuilder)
+
+    // use partial model for newData
+    val prediction = newData.connect(model).map(
+      (_: Int) => 0,
+      (_: Array[Double]) => 1
+    )
+
+    // emit result
+    if (params.has("output")) {
+      prediction.writeAsText(params.get("output"))
+    } else {
+      println("Printing result to stdout. Use --output to specify output path.")
+      prediction.print()
+    }
+
+    // execute program
+    env.execute("Streaming Incremental Learning")
+  }
+
+  // *************************************************************************
+  // USER FUNCTIONS
+  // *************************************************************************
+
+  /**
+    * Feeds new data for newData. By default it is implemented as constantly
+    * emitting the Integer 1 in a loop.
+    */
+  private class FiniteNewDataSource extends SourceFunction[Int] {
+    var counter: Int = 0
+
+    override def run(ctx: SourceContext[Int]) = {
+      Thread.sleep(15)
+      while (counter < 50) {
+        ctx.collect(getNewData)
+      }
+    }
+
+    def getNewData = {
+      Thread.sleep(5)
+      counter += 1
+      1
+    }
+
+    override def cancel() = {
+      // No cleanup needed
+    }
+  }
+
+  /**
+    * Feeds new training data for the partial model builder. By default it is
+    * implemented as constantly emitting the Integer 1 in a loop.
+    */
+  private class FiniteTrainingDataSource extends SourceFunction[Int] {
+    var counter = 0
+
+    override def run(ctx: SourceContext[Int]) = {
+      while (counter < 8200) ctx.collect(getTrainingData)
+    }
+
+    def getTrainingData = {
+      counter += 1
+      1
+    }
+
+    override def cancel() = {
+      // No cleanup needed
+    }
+  }
+
+  private class LinearTimestamp extends AssignerWithPunctuatedWatermarks[Int] {
+    var counter = 0L
+
+    override def extractTimestamp(element: Int, previousElementTimestamp: Long): Long = {
+      counter += 10L
+      counter
+    }
+
+    override def checkAndGetNextWatermark(lastElement: Int, extractedTimestamp: Long) = {
+      new Watermark(counter - 1)
+    }
+  }
+
+  /**
+    * Builds up-to-date partial models on new training data.
+    */
+  private class PartialModelBuilder extends AllWindowFunction[Int, Array[Double], TimeWindow] {
+    override def apply(window: TimeWindow,
+                       input: Iterable[Int],
+                       out: Collector[Array[Double]]): Unit = {
+      out.collect(Array[Double](1.0))
```

Comment:
Same here, the `buildPartialModel` function serves an illustrative purpose in the Java code.